packages/@ember/-internals/glimmer/lib/environment.ts

```diff
@@ -50,29 +51,36 @@ setGlobalContext({
       { id: 'ember-htmlbars.style-xss-warning' }
     );
   },
-});
 
-if (DEBUG) {
-  setTrackingTransactionEnv!({
-    assert(message) {
-      assert(message, false);
-    },
+  assert(test: unknown, msg: string, options?: { id: string }) {
+    if (DEBUG) {
+      let id = options?.id;
 
-    deprecate(message) {
-      deprecate(message, false, {
-        id: 'autotracking.mutation-after-consumption',
-        until: '4.0.0',
-        for: 'ember-source',
-        since: {
-          enabled: '3.21.0',
-        },
-      });
-    },
+      let override = VM_ASSERTION_OVERRIDES.find((o) => o.id === id);
```

Comment:
Can't use `.find` on IE11 (this is failing browser stack)

Reply:
I think we should probably change the overrides to be a map / pojo where the key is the `id`.  What do you think?